### PR TITLE
Bug/enable deferred encryption

### PR DESCRIPTION
**Pull request by andrejb**

### CODE REVIEW

File: scripts/db_access/client_side_db.py [hunk 1]
```diff
@@ -227,18 +244,41 @@ def _main(soledad, km, args):
...
         if args.export_incoming_messages:
             yield _export_incoming_messages(soledad, args.export_incoming_messages)
-    except:
-        pass
+    except Exception as e:
+        logger.error(e)
     finally:
-        reactor.stop()
+        soledad.close()
+        reactor.callWhenRunning(reactor.stop)
```
Comment (kalikaneko) [inline scripts/db_access/client_side_db.py]: I don't quite understand this. this shouldn't be a callback to any action? this way, we're stopping it as soon as it starts, does that makes sense? maybe reactor.callLater(1, reactor.stop) or something?

File: scripts/db_access/client_side_db.py [hunk 2]
```diff
@@ -227,18 +244,41 @@ def _main(soledad, km, args):
...
-    uuid, server_url, cert_file, token = \
-        _get_soledad_info(args.username, args.provider, passphrase, basedir)
+
+    if not args.use_auth_data:
+        # get auth data from server
+        uuid, server_url, cert_file, token = \
+            _get_soledad_info(args.username, args.provider, passphrase, basedir)
+    else:
+        # load auth data from file
+        with open(args.use_auth_data) as f:
```
Comment (kalikaneko) [inline scripts/db_access/client_side_db.py]: can you document this format somewhere, maybe in the module docs?

File: scripts/db_access/client_side_db.py (hunk 1 above)
Comment (andrejb) [inline scripts/db_access/client_side_db.py]: The main function has the `inlineCallbacks` decorator, so the `reactor.stop` callback will only be added after all main function yields have returned. Anyway, I don't see what would be a difference between the way it is proposed (using `callWhenRunning`) and using `callLater` with a delay of 1 second.

File: scripts/db_access/client_side_db.py (hunk 2 above)
Comment (andrejb) [inline scripts/db_access/client_side_db.py]: Done.

File: scripts/db_access/client_side_db.py (hunk 1 above)
Comment (kalikaneko) [inline scripts/db_access/client_side_db.py]: ah, I was looking on a quite short scope, now that I see the inlinecallbacks it make sense

**Comment by kalikaneko** [comment]
I rebased and merged this one, closing.

**Pull request closed without merging**